modules/table/src/main/java/org/apache/ignite/internal/table/distributed/storage/InternalTableImpl.java

```diff
@@ -2231,6 +2232,27 @@ public ScheduledExecutorService streamerFlushExecutor() {
         return streamerFlushExecutor.get();
     }
 
+    @Override
+    public CompletableFuture<Long> estimatedSize() {
+        var invokeFutures = new CompletableFuture<?>[partitions];
+
+        for (int partId = 0; partId < partitions; partId++) {
+            var replicaGroupId = new TablePartitionId(tableId, partId);
+
+            invokeFutures[partId] = partitionLocation(replicaGroupId)
```

Comment:
I'd rather implement it in a bit different manner. For each partition I'd call awaitPrimaryReplica() in order to define the location. Just like we do for any other request.